### DISCUSSION TITLE
CSS stylings not properly loading

### DISCUSSION THREAD

**Discussion by joshvarghese2008** [vercel/next.js#74178] [Help] [answered]
### Summary

Hey guys,

I've noticed issues that other people had on their deployments on Vercel, I have the same sort of issue. When I run my website locally, it looks perfectly normal, but when I deploy it, a lot of the css stylings seem to not work.

Any help is appreciated.  I'm doing this for a church website. I am using css and a bit of tailwind for the nav bar.

This is the website run locally
<img width="1396" alt="Screenshot 2024-12-20 at 18 08 04" src="https://github.com/user-attachments/assets/4f1d28a5-79aa-4f99-844a-bebe789ea078" />
<img width="1396" alt="Screenshot 2024-12-20 at 18 09 20" src="https://github.com/user-attachments/assets/0bd47da9-0255-4103-8e10-4905f9746dcc" />

This is the website after it's deployed
<img width="1396" alt="Screenshot 2024-12-20 at 18 08 13" src="https://github.com/user-attachments/assets/4f612a69-306e-4e17-a7a7-8df1eca631a9" />
<img width="1396" alt="Screenshot 2024-12-20 at 18 09 32" src="https://github.com/user-attachments/assets/58d0cc88-776f-4c99-a057-9c457227819f" />

**Comment by icyJoseph** [comment]
To me this site, https://stackblitz.com/github/joshvarghese2008/sample-website?file=README.md, looks as the problem you describe in production, but it is on dev mode.

**Reply by joshvarghese2008** [reply]
Hi,

I ran this on the website link you provided, and the same issue is occuring. What's the difference between dev and production, and how do I get it to work in production mode?

**Reply by icyJoseph** [reply]
I see it wrong in dev mode too...

**Reply by icyJoseph** [reply]
One issue could be, CSS ordering, but I am not sure, could you share a link to your prod site?

**Reply by joshvarghese2008** [reply]
https://sample-website-trial-2.vercel.app/

hi this is the site that I deployed on vercel

**Reply by joshvarghese2008** [reply]
> One issue could be, CSS ordering, but I am not sure, could you share a link to your prod site?

I believe this may be, I saw this was mentioned in another post. How do I go about this?

**Reply by icyJoseph** [reply]
Not much unfortunately, it should work out of the box.

Let's rethink, maybe the issue is minification?

- https://github.com/vercel/next.js/issues/7494

Do that to the webpack config in your next.config.js file

**Reply by joshvarghese2008** [reply]
quick question, is there meant to be any configs in the file,

right now it looks like this...

![image](https://github.com/user-attachments/assets/63091836-7f22-4fce-84bd-4f90931eb526)

**Reply by joshvarghese2008** [reply]
what should I add to this file?

**Reply by icyJoseph** [reply]
```js
const nextConfig = {
  // ...
  /**
   * @param {import('webpack').Configuration} webpackConfig
   * @returns {import('webpack').Configuration}
   */
  webpack(webpackConfig) {
    return {
      ...webpackConfig,
      optimization: {
        minimize: false,
      },
    };
  },
};

```

**Reply by joshvarghese2008** [reply]
hello,

i'm still facing the same styling issue.

in the deployed website its throwing these errors
<img width="580" alt="image" src="https://github.com/user-attachments/assets/79eaa04d-0890-4be4-8f3e-153b50a02195" />

on the styles that are not being properly applied, it's say that the property value is invalid
<img width="655" alt="image" src="https://github.com/user-attachments/assets/c129e9ce-b823-4a75-9710-e18bf305b1d8" />


however none of these appear when i'm running the website locally

**Reply by icyJoseph** [reply]
It has to be minification somehow, or postcss processing.

Like, for example a missing `;` at the end of your file, could be ok in dev, but when processed, minified and joint together with other CSS can cause issues.

We need to review your CSS to see which rule (block of CSS) is doing this.

Start by removing some parts and rebuild. Also try not to have comments, I somehow remember that they can cause issues. Let's doubt everything.

**Reply by icyJoseph** [reply]
You don't need to deploy over and over. Do `npm run build && npm run start` after you do changes.

**Reply by joshvarghese2008** [reply]
In the screenshot I provided, would the error in border radius result from that? (I'm new to this whole minimising concept) so I'm confused on how something that doesn't work when the surrounding code is formatted properly...

**Reply by icyJoseph** [reply, marked as answer]
I mean, that's wrong:

> all length values in CSS must be followed by a unit and the only exception to this is 0.

Use units on that one.

**Reply by joshvarghese2008** [reply]
I see... lemme try fixing that

**Reply by joshvarghese2008** [reply]
Hi this fixed the problem. I didn't realise that without the suffix the production build wouldn't add the necessary spacings and that..

Thank you for your help

**Reply by icyJoseph** [reply]
Awesome, please don't forget to mark as answered! Have a nice day.
